crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2329,6 +2329,8 @@ impl<'db> TypeInferenceBuilder<'db> {
             .expect("Symbol table should create a symbol for every Name node")
             .is_bound();
 
+        let mut ty = Type::Unbound;
```

Comment:
I think this change to the structure of the function is not necessary in this PR, and I have mixed feelings about it. One the one hand I see that it removes a couple `else` clauses and arguably makes the function structure easier to understand. On the other hand, I think it might move this function further away from the structure it will need to have when we remove `Type::Unbound`, at which point there won't be any reasonable "default" value to set `ty` to?

Reply:
Happy to revert this change, it was just a driveby thing to improve readability